events/validators.go

```diff
@@ -51,24 +83,32 @@ func ValidateEvent(name string, t EventType, got []byte) error {
 	case CloudEvent:
 		return validateCloudEvent(name, got, want)
 	}
+
+	// Should be unreachable.
 	return nil
 }
 
-func validateLegacyEvent(name string, gotBytes, wantBytes []byte) error {
+func validateLegacyEvent(name string, gotBytes, wantBytes []byte) *ValidationInfo {
+	vi := &ValidationInfo{
+		Name: name,
+	}
 	got := make(map[string]interface{})
 	err := json.Unmarshal(gotBytes, &got)
 	if err != nil {
-		return fmt.Errorf("unmarshalling function-received version of legacy event %q: %v", name, err)
+		vi.Errs = []error{fmt.Errorf("unmarshalling function-received version of legacy event %q: %v", name, err)}
```

Comment:
Append the error if getting an error when unmarshalling `got` and `want` and return if `vi.Errs` is not `nil` so we can check both?